### ISSUE TITLE
Opt out of generating Mocks for debugging target

### ISSUE THREAD

**Issue by markst** [Kolos65/Mockable#95]
We love `@Mockable` macro ❤️

However we only use it for unit tests. 
When used repeatedly throughout our codebase, the mocked definitions are generated by our target despite not being used.

We know we can use the `MOCKING` directive to remove the generated code from being compiled, however the macro will still generate the code using valuable processing power.

**Comment by Kolos65** [comment]
Do you see a significant compilation overhead in your project when using Mockable? I tested Mockable previously in very large projects and found that the expansion itself is extremely fast and does not affect build times, maybe I missed something...

The first idea I have is to avoid expansion is this:

```swift
#if MOCKING
@Mockable
#endif
protocol MyService {
    func fetch() async throws
}
```

**Comment by ben-p-commits** [comment]
Agreed... feels like you'd need a truly shocking quantity of protocols being mocked to see any sort of significant cost from `@Mockable`.

**Comment by markst** [comment]
This is @aduuub’s findings. I’ve asked him to share some benchmarks

**Comment by markst** [comment]
Will hopefully get some better benchmarks, but as a real quick assessment this is incremental build time without `@Mockable` definitions: 

<img width="327" alt="image" src="https://github.com/user-attachments/assets/db951d55-ff25-415d-b4a2-cd7f31c90c2c" />

vs with:

<img width="327" alt="image" src="https://github.com/user-attachments/assets/99a64e12-bd14-4846-a757-1b96edb0c91f" />


I'm not really sure there's anything to worry about...

**Comment by markst** [comment]
Closing this as it appears was a miscommunication.

**Comment by Kolos65** [comment]
@markst Here is a great acrticle on using macros at scale, you might find interesting: https://tuist.dev/blog/2024/08/26/swift-macros/
